### DISCUSSION TITLE
Migration Pod Error While Upgrading Chart Helm 1.3.0 to 1.5.0

### DISCUSSION THREAD

**Discussion by silastailer** [apache/airflow#23365] [General]
### Official Helm Chart version

1.5.0 (latest released)

### Apache Airflow version

2.1.4

### Kubernetes Version

1.20

### Helm Chart configuration

I will post the major differences between the current version that is working and the new configs:

```securityContext: {}```

`defaultAirflowTag: "2.2.4"`

`useDefaultImageForMigration: true`

`migrationsWaitTimeout: 120`

### Docker Image customisations

We extended the default image and install this libraries:

```
Authlib==0.15.4
apache-airflow-providers-slack==4.0.1
apache-airflow-providers-amazon==2.6.0
```

### What happened

When we try to build this new version in our pipeline we receive the following error:

`ERROR! Maximum number of retries (20) reached.
Last check result:
$ airflow db check
Traceback (most recent call last):
  File “/home/airflow/.local/bin/airflow”, line 5, in <module>
    from airflow.__main__ import main
  File “/home/airflow/.local/lib/python3.7/site-packages/airflow/__init__.py”, line 34, in <module>
    from airflow import settings
  File “/home/airflow/.local/lib/python3.7/site-packages/airflow/settings.py”, line 35, in <module>
    from airflow.configuration import AIRFLOW_HOME, WEBSERVER_CONFIG, conf  # NOQA F401
  File “/home/airflow/.local/lib/python3.7/site-packages/airflow/configuration.py”, line 1129, in <module>
    conf.validate()
  File “/home/airflow/.local/lib/python3.7/site-packages/airflow/configuration.py”, line 224, in validate
    self._validate_config_dependencies()
  File “/home/airflow/.local/lib/python3.7/site-packages/airflow/configuration.py”, line 267, in _validate_config_dependencies
    raise AirflowConfigException(f”error: cannot use sqlite with the {self.get(‘core’, ‘executor’)}“)
airflow.exceptions.AirflowConfigException: error: cannot use sqlite with the KubernetesExecutor`

### What you think should happen instead

We use a external database (rds postgres) configured in the chart inside section _metadataConnection_.

We are not understading why `$ airflow db check` is trying to run with SQLite as it should check the external one.

### How to reproduce

- Must have a external database for Airflow
- Must be using KubernetesExecutor
- Configure migrations parameters as they are now required.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by potiuk** [comment]
You have not written which component is producing this error.?

**Comment by potiuk** [comment]
BTW. Quick solution to that will be to set `CONNECTION_CHECK_MAX_COUNT` variable to `0` for the failing POD but it;'s not going to solve the problem becaue apparently for some reason the pod (I assume from the title) doing migration got a wrong configuration for SQL_ALCHEMY connection.

**Comment by potiuk** [comment]
Is it possible that you have `AIRFLOW__CORE__SQL_ALCHEMY_CONN`  defined separately for your appliation?
Could you export your failing job defintion from kubernetes using kubeectl ?

**Comment by silastailer** [comment]
Hello,
I will try your suggestion and get the information you asked.
Thanks a lot.

**Comment by gustavodamaceno1** [comment]
Hello,
We are having this issue in the pod run-migration-db,
CONNECTION_CHECK_MAX_COUNT is set as 0 (default value)
AIRFLOW__CORE__SQL_ALCHEMY_CONN: false
and in the section webserverConfig we import flask_appbuilder.security.manager import AUTH_OAUTH in order to configure Google Oath

**Comment by potiuk** [comment]
> AIRFLOW__CORE__SQL_ALCHEMY_CONN: false

What do you mean by "false"? This is a real connection you should have here?

**Reply by sudeepgupta90** [reply]
I think `AIRFLOW__CORE__SQL_ALCHEMY_CONN:false` is a reference to `enableBuiltInSecretEnvVars` in the Helm Chart

**Comment by potiuk** [comment]
BTW. I will convert it into discussion - until it's vleared - this is likely your problem with configuration you need to debug./

**Reply by sudeepgupta90** [reply]
I came across this after a year this conversation got started, and had the same error while I was testing a new version of helm chart. Like you suggested its a config problem which we found and fixed at our end. Nothing to be improved in chart or docs
